src/site/content/en/reliable/app-shell-ux-sw/index.md

```diff
@@ -0,0 +1,151 @@
+---
+layout: post
+title: App Shell UX with service workers and streams
+authors:
+  - demianrenzulli
+  - jeffposnick
+date: 2020-07-20
+description: |
+  How to achieve an SPA-like architecture in multi-page apps by combining partials, service workers, and streams.
+---
+
+[Single-page app (SPA)](https://en.wikipedia.org/wiki/Single-page_application) is an architectural pattern in which the browser runs JavaScript code to update the existing page when the user visits a different section of the site, as opposed to loading an entire new page.
+
+This means that the web app doesn't perform an actual page reload. The [History API](https://developer.mozilla.org/en-US/docs/Web/API/History_API) is used instead to navigate back and forth through the user's history and manipulate the contents of the history stack.
+
+Using this type of architecture can provide an [app shell UX](https://developers.google.com/web/fundamentals/architecture/app-shell) that's fast, reliable, and usually consumes less data when navigating.
+
+In multi-page apps (MPAs) each time a user navigates to a new URL, the browser progressively renders HTML specific to that page. This means a full page refresh every time you visit a new page.
+
+While both are equally valid models to use, you might want to bring some of the benefits of the app-shell UX of SPAs to your existing MPA site.
+In this article we'll analyze how you can achieve an SPA-like architecture in multi-page apps by combining partials, service workers, and streams.
+
+## Production case
+
+[DEV](https://dev.to/) is a platform where software developers write articles, take part in discussions, and build their professional profiles.
+
+Their architecture is a multi-page app based on traditional backend templating through Ruby on Rails. Their team was interested in some of the benefits of an app shell model, but didn't want to undertake a major architectural change or move away from their original tech stack. 
```

Comment:
```suggestion
[DEV](https://dev.to/) is a platform where software developers write articles, take part in discussions, and build their professional profiles.

<figure class="w-figure">
  <img class="w-screenshot w-screenshot--filled" 
       src="homepage.jpg" 
       alt="A screenshot of https://dev.to">
</figure>

Their architecture is a multi-page app based on traditional backend templating through Ruby on Rails. Their team was interested in some of the benefits of an app shell model, but didn't want to undertake a major architectural change or move away from their original tech stack. 
```

Please add this picture to your branch as `homepage.jpg`

![homepage](https://user-images.githubusercontent.com/4713486/85346936-e3a6fb00-b4ab-11ea-9b28-041bc84f63de.jpg)

Reply:
This is very nice, thank you for taking the time to take the screenshot.